presto-native-execution/presto_cpp/main/types/tests/PrestoToVeloxSplitTest.cpp

```diff
@@ -44,9 +45,15 @@ protocol::ScheduledSplit makeHiveScheduledSplit() {
 }
 } // namespace
 
-class PrestoToVeloxSplitTest : public ::testing::Test {};
+class PrestoToVeloxSplitTest : public ::testing::Test {
+ protected:
+  static void SetUpTestCase() {
```

Comment:
Let's use `void SetUp() override` and `void TearDown() override`

Reply:
If someone runs these tests in parallel using something like https://github.com/google/gtest-parallel, then the static function can cause problems.

Reply:
How does the gtest-parallel work? Does it start separate processes for different fixture with different test selected? If that's the case does it change anything?

I actually think there's an another problem and ideally the `registerPrestoToVeloxConnector` should be called not at the fixture (test suite) level, but on the binary level via a global setup (https://google.github.io/googletest/advanced.html#global-set-up-and-tear-down).

The `SetUpTestCase` is called once for every "fixture" (or test suite in other words) in a binary. A single test binary can be executing multiple fixtures. In theory if there's one more test added to the same binary that needs to register a HivePrestoToVeloxConnector  with the same name it may fail (as it may already be registered by an another fixture). But this is not the case today. The two tests that need the connector to be registered (`PrestoToVeloxSplitTest` and `PlanConverterTest`) are currently separated into two separate binaries. 

I didn't find us using global setup anywhere, and since the tests were already structured this way (`PrestoToVeloxSplitTest` and `PlanConverterTest` being in separate binaries) I decided not to change it. But it feels like ideally this initialization has to be moved even one level above.

Reply:
I see that this works today. But in general, I feel it is cleaner to keep the initialization local to a test than at the global level and hope there are no global conflicts. The example conflict you gave of adding another protocol instance by a subsequent test shows this. Let's stick to what other tests do to conclude this.

Reply:
Discussed offline. Going to add `unregister` as well as move it to `SetUp`  / `TearDown` due to: 

> Note that SetUpTestSuite() may be called multiple times for a test fixture class that has derived classes

https://google.github.io/googletest/advanced.html#global-set-up-and-tear-down